extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +124,194 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * Sends an Amp Analytics trigger event to a vendor's cross-domain iframe,
+   * or queues the message if the frame is not yet ready to receive messages.
+   * TODO: Implement throttling if messages are sent too rapidly.
+   * @param {string} request
+   * @param {!Object<string, string>} transportOptions
+   */
+  static sendRequestUsingCrossDomainIframe(request, transportOptions) {
+    const frameUrl = Transport.appendHashToUrl_(transportOptions['iframe'],
+      transportOptions['dataHash']);
+    const frameData = Transport.crossDomainFrames[frameUrl];
+    if (frameData.isReady) {
+      this.sendToCrossDomainIframe_(frameData.frame, [request]);
+    } else {
+      frameData.msgQueue.push(request);
+    }
+  }
+
+  /**
+   * Takes a URL and a data hash (the part that comes after the '#',
+   * optionally including the '#' itself) and concatenates them, adding the
+   * '#' if necessary
+   * @param {!string} url
+   * @param {string=} dataHash
+   * @return {!string}
+   * @private
+   */
+  static appendHashToUrl_(url, dataHash) {
+    if (dataHash) {
+      return url + (startsWith(dataHash, '#') ? dataHash : '#' + dataHash);
+    }
+    return url;
+  }
+
+  /**
+   * If iframe (and optionally dataHash as well) are specified in
+   * config/transport, check whether third-party iframe already exists, and if
+   * not, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {Function} processResponse An optional function to receive any
+   * response messages back from the cross-domain iframe
+   */
+  static processCrossDomainIframe(ampDoc, transportOptions, processResponse) {
+    user().assert(!(transportOptions['beacon'] || transportOptions['xhrpost'] ||
+      transportOptions['image']), 'Cross-domain frame cannot coexist with' +
+      ' other transport methods');
+    const frameUrl = Transport.appendHashToUrl_(transportOptions['iframe'],
+        transportOptions['dataHash']);
+    // If iframe doesn't exist for this iframe url (and data hash), create it.
+    if (Transport.hasCrossDomainFrame(frameUrl)) {
+      this.incrementUsageCount_(frameUrl);
+    } else {
+      const frame = Transport.createCrossDomainFrame(ampDoc, frameUrl);
+      ampDoc.body.appendChild(frame);
+    }
+    if (processResponse) {
+      window.addEventListener("message", (msg) => {
```

Comment:
message should be single quoted.  Make sure to run gulp lint and gulp check-types periodically